### PR TITLE
dev-vcs/monotone: bump to EAPI6, misc QA cleanups

### PR DESCRIPTION
**Pull request by axs-gentoo**
- Added a tmpfiles.d script so that /var/run/monotone (which
is actually /run/monotone on a tmpfs filesystem due to the
/var/run -> /run symlink) still exists on reboot.

- Moved patches from ${FILESDIR} to a tarball due to the patches
being too marge

- Bumped EAPI to 6 as EAPI4 is deprecated:
	- eapply instead of epatch
	- eapply_user now supported
	- eutils and toolchain-funcs eclasses no longer needed

Package-Manager: Portage-2.3.8, Repoman-2.3.1

### CODE REVIEW

File: dev-vcs/monotone/monotone-1.0-r5.ebuild [hunk 1]
```diff
@@ -2,23 +2,24 @@
...
 
 DESCRIPTION="Monotone Distributed Version Control System"
 HOMEPAGE="http://monotone.ca"
-SRC_URI="http://monotone.ca/downloads/${PV}/${P}.tar.bz2"
+SRC_URI="http://monotone.ca/downloads/${PV}/${P}.tar.bz2
+	http://dev.gentoo.org/~axs/distfiles/${P}-patches.tar.xz"
 
 LICENSE="GPL-2"
 SLOT="1"
 KEYWORDS="amd64 ~ia64 x86"
```
Comment (jonasstein) [inline dev-vcs/monotone/monotone-1.0-r5.ebuild]: need all ~, if you bump EAPI

Comment (axs-gentoo) [reply dev-vcs/monotone/monotone-1.0-r5.ebuild]: Why??  Files-on-disk isn't changing (except for the /var/run permanency fix), just VDB, and this is a stable EAPI so nothing to worry about there either...

**Comment by monsieurp** [comment]
@axs-gentoo ping

**Comment by axs-gentoo** [comment]
pong, still waiting for a valid reason for all ~arch and a stabilization process just because EAPI was bumped.

**Comment by kensington** [comment]
I think the argument is that the ebuild behaviour can vary wildly between versions. For example, I've seen the past seemingly simple EAPI bumps that caused the package to not even build anymore.

**Comment by mgorny** [comment]
Please remember that eclasses have some EAPI-conditional code, and it's hard to predict what side effects an EAPI bump might have. In any case, AFAICS this is maintainer-needed, so do as you see fit.

**Comment by flappyports** [comment]
dev-vcs/monotone was list-rited and removed in 2a13d51aa30ee1554239cbc8655f5aab12e08036

**Pull request closed without merging**